src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/NodeMonitoringFailoverPlugin.java

```diff
@@ -221,16 +228,37 @@ protected void initNodeKeys(Connection connection) {
           "[NodeMonitoringFailoverPlugin.initNodes]: Could not retrieve Host:Port from querying");
     }
 
-    nodeKeys.add(
+    this.nodeKeys.add(
         String.format("%s:%s",
-            this.hostInfo.getHost(),
-            this.hostInfo.getPort()
+            hostInfo.getHost(),
+            hostInfo.getPort()
         ));
+
+    return this.nodeKeys;
   }
 
   private void assertArgumentIsNotNull(Object param, String paramName) {
     if (param == null) {
       throw new NullArgumentException(paramName);
     }
   }
+
+  /**
+   * Check if the connection has changed due to failover.
+   * If so, clear the set of keys for the dead node
+   * and remove monitor's references to that node.
+   *
+   * @param newConnection The connection used by {@link ClusterAwareConnectionProxy}.
+   * @return the most up-to-date connection.
+   */
+  private Connection validateConnection(Connection newConnection) {
+    final boolean isSameConnection = this.connection.equals(newConnection);
+    if (!isSameConnection) {
+      this.monitorService.failoverMonitor(this.nodeKeys);
```

Comment:
Unclear of what this does, knowing that you clear nodeKeys right after

Reply:
Discussed offline.